src/main/host/process.c

```diff
@@ -496,10 +469,11 @@ static void _unref_thread_cb(gpointer data) {
 
 void process_addThread(Process* proc, Thread* thread) {
     MAGIC_ASSERT(proc);
-    g_hash_table_insert(proc->threads, GUINT_TO_POINTER(thread_getID(thread)), thread);
+    _process_insertThread(proc->rustProcess, thread);
 
-    // Schedule thread to start.
-    thread_ref(thread);
+    // Schedule thread to start. We're giving the caller's reference to thread
+    // to the TaskRef here, which is why we don't increment its ref count to
+    // create the TaskRef, but do decrement it on cleanup.
```

Comment:
I don't see a change to the caller of `process_addThread`, so was this incorrect before? Or has something unrelated changed that affects the caller?

Reply:
`_process_insertThread` increments the reference count